MinimalCirclesPath.py

```diff
@@ -0,0 +1,99 @@
+from math import sin, cos, radians, asin, atan2, sqrt, degrees, ceil, floor
+
+def calculateBearing(startLat, startLon, destLat, destLon):
+    startLatRad = radians(startLat)
+    startLonRad = radians(startLon)
+    destLatRad = radians(destLat)
+    destLonRad = radians(destLon)
+
+    deltaLon = destLonRad - startLonRad
+
+    y = sin(deltaLon) * cos(destLatRad)
+    x = cos(startLatRad) * sin(destLatRad) - sin(startLatRad) * cos(destLatRad) * cos(deltaLon)
+
+    initialBearingRad = atan2(y, x)
+
+    initialBearingDeg = degrees(initialBearingRad)
+    compassBearing = (initialBearingDeg + 360) % 360
+
+    return compassBearing
+
+def nextPoint(startLat, startLong, bearingDegrees, distanceNm=100):
+
+    earthRadiusNm = 3440.065  # Radius of the Earth in nautical miles
+
+    # Convert latitude and longitude to radians
+    startLanRad = radians(startLat)
+    startLongRad = radians(startLong)
+    bearingRad = radians(bearingDegrees)
+
+    # Calculate angular distance
+    angularDistanceRad = distanceNm / earthRadiusNm
+
+    # Calculate destination latitude
+    destLatRad = asin(sin(startLanRad) * cos(angularDistanceRad) +
+                         cos(startLanRad) * sin(angularDistanceRad) * cos(bearingRad))
+
+    # Calculate destination longitude
+    destLongRad = startLongRad + atan2(sin(bearingRad) * sin(angularDistanceRad) * cos(startLanRad),
+                                     cos(angularDistanceRad) - sin(startLanRad) * sin(destLatRad))
+
+    # Convert latitude and longitude back to degrees
+    destLat = degrees(destLatRad)
+    destLong = degrees(destLongRad)
+
+    return destLat, destLong
+
+# used geeksforgeeks.org
+def getDistance(startLat, startLong, destLat, destLong):
+    earthRadiusNm = 3440.065  # Radius of the Earth in nautical miles
+
+    # Convert latitude and longitude from degrees to radians
+    startLat, startLong, destLat, destLong = map(radians, [startLat, startLong, destLat, destLong])
+
+    # Calculate differences
+    latDifference = destLat - startLat
+    longDifference = destLong - startLong
+
+    # Calculate distance using Haversine formula
+    a = sin(latDifference / 2) ** 2 + cos(startLat) * cos(destLat) * sin(longDifference / 2) ** 2
+    c = 2 * asin(sqrt(a))
+    distanceNm = earthRadiusNm * c
+
+    return distanceNm
+
+# pathWidth is the desired scaned distance from play diameter, radius is radius of circle passed to NOTAM
+def getPath( startLat, startLong, destLat, destLong, radius, pathWidth):
+    # TODO For actual application you would need to capture N/S/E/W
+
+    # list of coordinates returned to call API with
+    coordList = [(startLat, startLong)]
+      
+    # gets total distance from start to finish
+    totalDistance = getDistance(startLat, startLong, destLat, destLong)
+
+    # gets step distance (pythagorean theorum)
+    stepDistance = sqrt((radius)**2-(pathWidth/2)**2)
```

Comment:
I'm confused why pathWidth is a factor in computer the step distance.  It looks like this is computing the long side of a right triangle where the other two sides are the radius and pathWidth/2 ... not sure I understand why

Reply:
my logic is that there is a right triangle from the starting point, to the half way point between the two calls, and then the point where the path and both circles intersect. Im attaching a photo but I think I forgot to multiple the step distance by 2
![tempImagea0Ur87](https://github.com/Gabrieldowen/NOTAMyWatch/assets/94488674/ec638aec-c1b8-422c-947c-0c65cac6f300)

Reply:
that makes sense, thanks

you'll find that details like this are worth documenting as inevitably someone down the road a few weeks (or months) will look at this code and have the same question yet again

Reply:
Your welcome!